gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/TraversalEvaluation/TraversalParser.cs

```diff
@@ -0,0 +1,472 @@
+        {
+            var firstChar = text[i];
+            while (char.IsWhiteSpace(firstChar))
+            {
+                firstChar = text[++i];
+            }
+            if (firstChar == ')')
+            {
+                return null;
+            }
+            if (firstChar == '"' || firstChar == '\'')
+            {
+                return StringParameter.Parse(text, firstChar, ref i);
+            }
+            if (char.IsDigit(firstChar))
+            {
+                return ParseNumber(text, ref i);
+            }
+            if (text.Substring(i, 3).StartsWith("__."))
+            {
+                var startIndex = i;
+                var tokens = ParseTokens(text, ref i);
+                return new StaticTraversalParameter(tokens, text.Substring(startIndex, i - startIndex));
+            }
+            if (text.Substring(i, 2).StartsWith("P."))
+            {
+                return new TraversalPredicateParameter(ParseTokens(text, ref i));
+            }
+            var parameterText = text.Substring(i, text.IndexOf(')', i) - i);
+            var separatorIndex = parameterText.IndexOf(',');
+            if (separatorIndex >= 0)
+            {
+                parameterText = parameterText.Substring(0, separatorIndex);
+            }
+            parameterText = parameterText.Trim();
+            if (parameterText == "")
+            {
+                return null;
+            }
+            if (parameterText == "true" || parameterText == "false")
+            {
+                i += parameterText.Length - 1;
+                return LiteralParameter.Create(Convert.ToBoolean(parameterText));
+            }
+            if (RegexEnum.IsMatch(parameterText))
+            {
+                i += parameterText.Length - 1;
+                return new TraversalEnumParameter(parameterText);
+            }
+            if (RegexParam.IsMatch(parameterText))
+            {
+                i += parameterText.Length - 1;
+                return new ContextBasedParameter(parameterText);
+            }
+            throw new NotSupportedException($"Parameter {parameterText} not supported");
+        }
+        
+        private static ITokenParameter ParseNumber(string text, ref int i)
+        {
+            var match = RegexNumeric.Match(text, i);
+            if (!match.Success)
+            {
+                throw new InvalidOperationException(
+                    $"Could not parse numeric value from the beginning of {text.Substring(i)}");
+            }
+            var numericText = match.Value.ToUpper();
+            i += match.Value.Length - 1;
+            if (numericText.EndsWith("L"))
+            {
+                return LiteralParameter.Create(Convert.ToInt64(match.Value.Substring(0, match.Value.Length - 1)));
+            }
+            if (numericText.EndsWith("F"))
+            {
+                return LiteralParameter.Create(Convert.ToSingle(match.Value.Substring(0, match.Value.Length-1)));
```

Comment:
Add `CultureInfo.InvariantCulture` for the `Convert.ToSingle` call to avoid problems when the local culture uses a different decimal mark.

Reply:
Nice catch! I'll push a fix for it.